cheroot/ssl/builtin.py

```diff
@@ -247,14 +247,18 @@ def __init__(
             private_key_password=private_key_password,
         )
 
+        key_password = private_key_password
+        if callable(private_key_password):
+            key_password = private_key_password()
+
         self.context = ssl.create_default_context(
             purpose=ssl.Purpose.CLIENT_AUTH,
             cafile=certificate_chain,
         )
         self.context.load_cert_chain(
             certificate,
             private_key,
-            password=private_key_password,
+            password=key_password,
```

Comment:
According to https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain, this may be a callable too. Perhaps, we could reuse the one from the other adapter? We could put it into a `._pem_processing` module, for example use as the default here, providing argument defaults via `functools.partial()`, for example.